idx/memory/memory.go

```diff
@@ -925,6 +925,8 @@ func (m *UnpartitionedMemoryIdx) add(archive *idx.Archive) {
 	path := def.NameWithTags()
 
 	if TagSupport {
+		sort.Strings(def.Tags)
```

Comment:
Is there a case where MetricData.SetId() has not been called prior to this line being executed? MetricData.SetId() sorts the tags. I thought we had added this before and then removed it.

Reply:
There are two callpaths leading to `.add()`. 
* When loading the index from the backend store, we directly add it to the index without calling `SetId()`. Of course we can assume that these entries have been written by MT, but we don't really know what users may do with their stores, so I'd prefer to not rely on that.
* When ingesting data:
  * When ingesting from kafka, with or without write queue, `.SetId()` does not get called. We could assume that the producer always sorts the strings, but I don't think we should rely on that.
  * When ingesting from carbon we do call `.SetId()`, so that's fine

I think it's also worth considering that if a slice is already sorted, then calling `sort.Strings()` on it should be very cheap. So I think just to be not have to rely on assumptions, this is still worth it.

Reply:
Assuming that this is not a blocker, I'll already go ahead and merge. We can still discuss removing it again after merging.